tests/integration/targets/clickhouse_quota/tasks/initial.yml

```
@@ -0,0 +1,406 @@
+    limits:
+      - interval: 5 minute
+        randomized_start: true
+        max:
+          execution_time: 10.2
+      - interval: 1 day
+        max:
+          query_selects: 10
+          query_inserts: 10
+    apply_to: CURRENT_USER
+
+- name: Test 8 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is changed
+    - result.executed_statements == ['CREATE QUOTA \'test_quota\' ON CLUSTER \'test_cluster\' KEYED BY user_name FOR RANDOMIZED INTERVAL 5 minute MAX execution_time = 10.2, FOR INTERVAL 1 day MAX query_selects = 10, query_inserts = 10 TO CURRENT_USER']
+
+- name: Test 8 - Check the role was not created in check mode
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"
+
+- name: Test 8 - Verify that the role is not present
+  ansible.builtin.assert:
+    that:
+    - result.result == []
+
+# Test 9 - Negative test that you can't specify apply_to_mode == all and a list of users
+- name: Test 9 - Negative test that you can't specify apply_to_mode == all and a list of users
+  register: result
+  check_mode: true
+  ignore_errors: true
+  community.clickhouse.clickhouse_quota:
+    state: present
+    name: test_quota
+    apply_to: CURRENT_USER
+    apply_to_mode: all
+
+- name: Test 9 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is failed
+
+# Test 10 - Actual test with basic limits
+- name: Test 10 - Actual test with basic limits
+  register: result
+  community.clickhouse.clickhouse_quota:
+    name: test_quota
+    keyed_by: user_name
+    limits:
+      - interval: 5 minute
+        randomized_start: true
+        max:
+          queries: 100
+    apply_to: default
+
+- name: Test 10 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is changed
+    - result.executed_statements == ['CREATE QUOTA \'test_quota\' KEYED BY user_name FOR RANDOMIZED INTERVAL 5 minute MAX queries = 100 TO default']
+
+- name: Test 10 - Check the role was actually created
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"
+
+- name: Test 10 - Verify that the role is present
+  ansible.builtin.assert:
+    that:
+    - result.result == [['test_quota']]
+
+- name: Test 10 - Actually delete the quota
+  register: result
+  community.clickhouse.clickhouse_quota:
+    name: test_quota
+    state: absent
+
+- name: Test 10 - Check the return values
+  ansible.builtin.assert:
+    that:
+    - result is changed
+    - result.executed_statements == ['DROP QUOTA \'test_quota\'']
+
+- name: Test 10 - Check the role was actually deleted
+  register: result
+  community.clickhouse.clickhouse_client:
+    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"
+
+- name: Test 10 - Verify that the role is no longer present
+  ansible.builtin.assert:
+    that:
+    - result.result == []
```

```suggestion
- name: Test 10 - Actually delete the quota
  register: result
  community.clickhouse.clickhouse_quota:
    name: test_quota
    state: absent

- name: Test 10 - Check the return values
  ansible.builtin.assert:
    that:
    - result is changed
    - result.executed_statements == ['DROP QUOTA \'test_quota\'']

- name: Test 10 - Check the role was actually deleted
  register: result
  community.clickhouse.clickhouse_client:
    execute: "SELECT name FROM system.quotas WHERE name = 'test_quota'"

- name: Test 10 - Verify that the role is no longer present
  ansible.builtin.assert:
    that:
    - result.result == []
```
- We should also do this in check mode
- That's probably why we shouldn't mark tasks with `# Test NUM..` because can be painful to add something later, we'll sometimes have to re-number all the tasks after. I'd suggest just describing what a task does w/o numbering